src/voucherPurchase/Confirmation.js

```diff
@@ -0,0 +1,41 @@
+import React, { Component } from 'react';
+import voucherApi from '../services/voucherApi'
+import Voucher from './Voucher'
+import './Confirmation.css'
+
+class Confirmation extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {voucher: null}
+  }
+
+  componentDidMount = async () => {
+    if (this.props.post === true){
+    const qs = require('query-string')
+    const amount = qs.parse(this.props.location.search).amount
+    const response = await voucherApi.postVoucher(amount);
+    this.props.history.push("/thank-you?code=" + response.code)}
+    else {
+      const qs = require('query-string')
+      const code = qs.parse(this.props.location.search).code
+      const response = await voucherApi.getVoucher(code);
+      this.setState({ voucher: response})
+    }
+  }
+  
```

Comment:
Maybe extract the sections within if and else into own methods to make it clearer what they are doing